entrypoint.sh

```diff
@@ -100,37 +99,139 @@ function super_lint() {
   /action/lib/linter.sh
 }
 
+function get_files_matching_filetype() {
+  local filetype=$1
+  shift
+
+  matching_files=()
+
+  for file in "$@"; do
+    filename=$(basename "$file")
+    
+    if [[ "$filename" == *"$filetype" ]]; then
+      matching_files+=("$file")
+    fi
+  done
+  echo "${matching_files[@]}"
+}
+
+function linter_failed() {
+  return=$1
+  shift 
+  name=$1
+
+  if [[ "${return:-1}" != 0 ]]; then
+    linter_exit_codes+=(["$name"]="${return}")
+  else
+    return "${return}"
+  fi
+}
+
+function lint_files() {
+  exec &>> "${linter[logfile]}"
+  if [[ "${linter[filetype]}" = "all" ]]; then
+    cmd="${linter[name]} ${linter[args]}"
+    eval "$cmd"
+  else
```

Comment:
Instead of this else we could just return, then `fi`, and have the following `for` loop one level out-dented for readability.  <3 guard clauses